doc/upstream_release_process.md

```diff
@@ -1,152 +1,163 @@
-proceed.  Ensure that your local master is on that squash-merged commit
-before proceeding.
-
-## Update Release info on Launchpad.
-Go to https://launchpad.net/cloud-init click the milestone that we're
-releasing.  That will take you to
-[lp/cloud-init/+milestone/<X.Y>](http://launchpad.net/cloud-init/+milestone/17.2)
-.  Hit 'Create release'.
-
- Model the Release notes after other releases such as
- [17.1](https://launchpad.net/cloud-init/+milestone/17.1/) or
- [17.2](https://launchpad.net/cloud-init/+milestone/17.2)
-
-To help get some of those bits of information, add '| wc -l' for just
-the numbers. not done here to have you sanity check output.
-
-  * just get log into a file
-
-        git log 18.2..HEAD > git-log
-
-  * contributors
-
-        grep Author git-log | sed 's,.*: ,,'  | sort -u
-
-  * top level domains
-
-        grep Author git-log | sed 's,.*: ,,'  | sort -u | sed 's,.*@,,' | sort -u
-
-  * bugs
-
-        grep '^[ ]*LP' git-log | sed -e 's,#,,g' -e 's,.*LP: ,,' -e 's/,[ ]*/\n/'
-
-## Tag and push it to git
-
-First, ensure that you have a clean working tree with the squash-merged
-release commit:
-
-    git checkout master
-    git fetch upstream
-    git reset --hard upstream/master
-
-Then create a signed tag, pointing at the squash-merged release commit:
-
-    git tag --annotate --sign YY.N
-
+This document assumes the Canonical cloud-init remote is named `upstream` whereas your personal fork is named `origin`. For example,
+```bash
+$ git remote -v
+origin	git@github.com:MyGithubName/cloud-init.git (fetch)
+origin	git@github.com:MyGithubName/cloud-init.git (push)
+upstream	git@github.com:canonical/cloud-init.git (fetch)
+upstream	git@github.com:canonical/cloud-init.git (push)
+```
+
+Adjust any references to `upstream` and `origin` accordingly if yours are different.
+
+# Pre-release
+## Send pre-release email to mailing list
+Send an email to the cloud-init mailing list announcing the upcoming release. See previous emails for examples:
+* https://lists.launchpad.net/cloud-init/msg00357.html
+* https://lists.launchpad.net/cloud-init/msg00335.html
+
+## Perform pre-release checks
+* All outstanding merge proposals have been reviewed for merge
+* CI is green
+
+## Start upstream release bug
+Start creating a process bug in the cloud-init project that captures the new release version, highlights, and changelog.
+
+Use the `upstream-release` script (described in the next section) to generate the bug contents for you.
+
+See [previous release bugs in Launchpad](https://bugs.launchpad.net/cloud-init/+bugs?field.searchtext=Release&orderby=-datecreated&search=Search&field.status%3Alist=FIXRELEASED&field.importance%3Alist=UNDECIDED).
+
+
+## Create upstream-release branch
+**SAVE ALL LOCAL CHANGES TO A BRANCH**, then
+```bash
+git fetch upstream
+git checkout main
+git reset upstream/main --hard
+```
+
+Run `uss-tableflip/scripts/upstream-release <release_version> <old_version>` from cloud-init tree with updated main branch.
```

Comment:
The doc doesn't mention where to find this. Also some of those scripts assume that the other scripts are in PATH, e.g. `upstream-release` has:
```
NUM_BUGS=`git log $PREVIOUS_MAJOR_MINOR..HEAD | log2dch | grep "LP: #" | wc -l`
CHANGELOG=`git log $PREVIOUS_MAJOR_MINOR..HEAD | log2dch |  sed 's/^   //g'`
```
where `log2dch` is a `uss-tableflip` script. What I do is normally:
```
PATH=~/path/to/uss-tableflip/scripts:$PATH <script>
```
but symliking them to `~/bin` should also work.

Reply:
Thanks, I think I'll add a note at the top specifying which repositories are needed.